aptos-move/framework/src/natives/cryptography/algebra/mod.rs

```diff
@@ -58,6 +60,33 @@ pub enum Structure {
     BLS12381G2,
     BLS12381Gt,
     BLS12381Fr,
+    BN254(BN254Structure),
+}
+#[derive(Copy, Clone, Eq, Hash, PartialEq)]
+pub enum BN254Structure {
+    BN254Fr,
```

Comment:
As for the BN254 structure, at first, I just added them into `Structure` like you suggested.
But It seems also odd, because BN254Fr,Fq  have nothing to do with BLS12381Fr,Fq, so I abstracted a new BN254Structure, and had also thought abstracting a BLS12381Structure, and became something like:
```
enum Structure {
  Bls12381(Bls12381Structure),
  BN254(BN254Structure),
```

then we can handle each structure accordingly, it may also make abstraction easier, such as macro them all, or turn Feature a trait so we can reuse most of the code, and make adding new curve simple.
But for now, It's still not clear which way to go, and how to abstract them, after all, all we have for now are the two similar curves.

So  for now, it may be better to stay put, and make changes based on the ongoing situations.